ToolBox/IQFeed/IQFeedDataQueueUniverseProvider.cs

```diff
@@ -129,35 +129,35 @@ public Symbol GetLeanSymbol(string ticker, SecurityType securityType, string mar
         /// <param name="securityCurrency">Expected security currency(if any)</param>
         /// <param name="securityExchange">Expected security exchange name(if any)</param>
         /// <returns></returns>
-        public IEnumerable<Symbol> LookupSymbols(string lookupName, SecurityType securityType, bool includeExpired, string securityCurrency = null, string securityExchange = null)
+        public IEnumerable<Symbol> LookupSymbols(Symbol symbol, bool includeExpired, string securityCurrency = null)
         {
             Func<Symbol, string> lookupFunc;
 
-            switch (securityType)
+            switch (symbol.SecurityType)
             {
                 case SecurityType.Option:
                     // for option, futures contract we search the underlying
-                    lookupFunc = symbol => symbol.HasUnderlying ? symbol.Underlying.Value : string.Empty;
+                    lookupFunc = lookupSymbol => lookupSymbol.HasUnderlying ? lookupSymbol.Underlying.Value : string.Empty;
                     break;
                 case SecurityType.Future:
-                    lookupFunc = symbol => symbol.ID.Symbol;
+                    lookupFunc = lookupSymbol => lookupSymbol.ID.Symbol;
                     break;
                 default:
-                    lookupFunc = symbol => symbol.Value;
+                    lookupFunc = lookupSymbol => lookupSymbol.Value;
                     break;
             }
 
+            var lookupName = lookupFunc(symbol);
             var result = _symbolUniverse.Where(x => lookupFunc(x.Symbol) == lookupName &&
-                                            x.Symbol.ID.SecurityType == securityType &&
-                                            (securityCurrency == null || x.SecurityCurrency == securityCurrency) &&
-                                            (securityExchange == null || x.SecurityExchange == securityExchange))
+                                            x.Symbol.ID.SecurityType == symbol.SecurityType &&
+                                            (securityCurrency == null || x.SecurityCurrency == securityCurrency))
```

Comment:
Since IQFeed seems unrelated, wonder why the check changed, same bellow?

Reply:
Change of the interface required updates to the code. Thinking about it, I think I might be able to revert these changes and add a simple binding to the pre-existing method to have interface compatibility.